src/main/java/org/hmhb/program/published/PublishedProgramService.java

```diff
@@ -0,0 +1,17 @@
+package org.hmhb.program.published;
+
+import javax.annotation.Nonnull;
+
+import java.util.List;
+
+public interface PublishedProgramService {
+
+    PublishedProgram getById(@Nonnull Long id);
+
+    List<PublishedProgram> getAll();
+
+    PublishedProgram delete(@Nonnull Long id);
```

Comment:
I'm not certain if I've made the right choice on "long" vs. "@Nonnull Long" yet.  In the REST controllers, the param is "long" b/c it's a path param and it wouldn't hit that route unless there was an ID.

However, I'm not sure if the other callers of the service will elegantly check the non-null before calling or not (probably, but I'm not sure).